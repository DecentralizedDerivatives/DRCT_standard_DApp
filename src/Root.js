import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import rootReducer from './reducers';

const middlewares = [thunk];
let composeEnhancer = compose;

if (process.env.NODE_ENV === 'development') {
  middlewares.push(createLogger());
  composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
}

const Root = ({ children, initialState = {} }) => {
  const store = createStore(
    rootReducer,
    initialState,
    composeEnhancer(applyMiddleware(...middlewares))
  );

  return <Provider store={store}>{children}</Provider>;
};

export default Root;
