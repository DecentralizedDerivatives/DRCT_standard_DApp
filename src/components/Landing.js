import React, { Component } from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import '../styles/Landing.css';

class Landing extends Component {
  componentDidMount() {
    if (this.props.isConnectedMetamask && this.props.isConnectedNetwork) {
      this.props.history.push('/portfolio');
    }
  }

  render() {
    return (
      <div className="landing__bg">
        <div className="landing__overlay text-light">
          <div className="container">
            <div className="row">
              <div className="col-md-12 text-center">
                <h1 className="display-3 mb-4">DRCT DApp</h1>
                <p className="lead">
                  Please log into your MetaMask extension to access this site.
                </p>
                <hr />
                <a
                  className="btn btn-lg btn-light"
                  href="https://metamask.io/"
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  Need MetaMask?
                </a>

                <a
                  href="http://wwww.ddacoop.org"
                  className="btn btn-lg btn-info mr-2"
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  Learn More About DRCT DApp
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

Landing.propTypes = {
  isConnectedMetamask: PropTypes.bool.isRequired,
  isConnectedNetwork: PropTypes.bool.isRequired
};

const mapStateToProps = state => ({
  isConnectedMetamask: state.status.isConnectedMetamask,
  isConnectedNetwork: state.status.isConnectedNetwork
});

export default connect(mapStateToProps)(Landing);
