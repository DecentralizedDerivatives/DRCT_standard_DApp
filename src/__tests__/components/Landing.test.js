import React from 'react';
import { shallow } from 'enzyme';
import Root from '../../Root';

import Landing from '../../components/Landing';

describe('<Landing />', () => {
  describe('render()', () => {
    it('renders the component', () => {
      const wrapper = shallow(
        <Root>
          <Landing />
        </Root>
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
