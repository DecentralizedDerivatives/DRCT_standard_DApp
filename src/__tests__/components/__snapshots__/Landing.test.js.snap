// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Landing /> render() renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Landing
    dispatch={[Function]}
    metamask={false}
    network={0}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
    whiteListed={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="landing-head"
        >
          Decentralized Derivatives Association  DRCT
        </h1>,
        <h3
          className="landing-subhead"
        >
          <div>
            <p>
              Please Login to 
              <a
                href="https://metamask.io/"
              >
                MetaMask
              </a>
            </p>
            <p
              className="landing_p"
            >
              <a
                href="https://www.cryptocompare.com/wallets/guides/how-to-use-metamask/"
              >
                What is MetaMask?
              </a>
            </p>
            <div
              className="landing-network-status"
            >
              Select :
              <p
                className="rinkeby-network-label"
              >
                rinkeby test network
              </p>
            </div>
          </div>
        </h3>,
        <img
          alt=""
          className="landing-logo"
          src="logo-large.png"
        />,
      ],
      "id": "landing",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Decentralized Derivatives Association  DRCT",
          "className": "landing-head",
        },
        "ref": null,
        "rendered": "Decentralized Derivatives Association  DRCT",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            <p>
              Please Login to 
              <a
                href="https://metamask.io/"
              >
                MetaMask
              </a>
            </p>
            <p
              className="landing_p"
            >
              <a
                href="https://www.cryptocompare.com/wallets/guides/how-to-use-metamask/"
              >
                What is MetaMask?
              </a>
            </p>
            <div
              className="landing-network-status"
            >
              Select :
              <p
                className="rinkeby-network-label"
              >
                rinkeby test network
              </p>
            </div>
          </div>,
          "className": "landing-subhead",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Please Login to 
                <a
                  href="https://metamask.io/"
                >
                  MetaMask
                </a>
              </p>,
              <p
                className="landing_p"
              >
                <a
                  href="https://www.cryptocompare.com/wallets/guides/how-to-use-metamask/"
                >
                  What is MetaMask?
                </a>
              </p>,
              <div
                className="landing-network-status"
              >
                Select :
                <p
                  className="rinkeby-network-label"
                >
                  rinkeby test network
                </p>
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Please Login to ",
                  <a
                    href="https://metamask.io/"
                  >
                    MetaMask
                  </a>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Please Login to ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "MetaMask",
                    "href": "https://metamask.io/",
                  },
                  "ref": null,
                  "rendered": "MetaMask",
                  "type": "a",
                },
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href="https://www.cryptocompare.com/wallets/guides/how-to-use-metamask/"
                >
                  What is MetaMask?
                </a>,
                "className": "landing_p",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "What is MetaMask?",
                  "href": "https://www.cryptocompare.com/wallets/guides/how-to-use-metamask/",
                },
                "ref": null,
                "rendered": "What is MetaMask?",
                "type": "a",
              },
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Select :",
                  <p
                    className="rinkeby-network-label"
                  >
                    rinkeby test network
                  </p>,
                ],
                "className": "landing-network-status",
              },
              "ref": null,
              "rendered": Array [
                "Select :",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "rinkeby test network",
                    "className": "rinkeby-network-label",
                  },
                  "ref": null,
                  "rendered": "rinkeby test network",
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "className": "landing-logo",
          "src": "logo-large.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="landing-head"
          >
            Decentralized Derivatives Association  DRCT
          </h1>,
          <h3
            className="landing-subhead"
          >
            <div>
              <p>
                Please Login to 
                <a
                  href="https://metamask.io/"
                >
                  MetaMask
                </a>
              </p>
              <p
                className="landing_p"
              >
                <a
                  href="https://www.cryptocompare.com/wallets/guides/how-to-use-metamask/"
                >
                  What is MetaMask?
                </a>
              </p>
              <div
                className="landing-network-status"
              >
                Select :
                <p
                  className="rinkeby-network-label"
                >
                  rinkeby test network
                </p>
              </div>
            </div>
          </h3>,
          <img
            alt=""
            className="landing-logo"
            src="logo-large.png"
          />,
        ],
        "id": "landing",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Decentralized Derivatives Association  DRCT",
            "className": "landing-head",
          },
          "ref": null,
          "rendered": "Decentralized Derivatives Association  DRCT",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div>
              <p>
                Please Login to 
                <a
                  href="https://metamask.io/"
                >
                  MetaMask
                </a>
              </p>
              <p
                className="landing_p"
              >
                <a
                  href="https://www.cryptocompare.com/wallets/guides/how-to-use-metamask/"
                >
                  What is MetaMask?
                </a>
              </p>
              <div
                className="landing-network-status"
              >
                Select :
                <p
                  className="rinkeby-network-label"
                >
                  rinkeby test network
                </p>
              </div>
            </div>,
            "className": "landing-subhead",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Please Login to 
                  <a
                    href="https://metamask.io/"
                  >
                    MetaMask
                  </a>
                </p>,
                <p
                  className="landing_p"
                >
                  <a
                    href="https://www.cryptocompare.com/wallets/guides/how-to-use-metamask/"
                  >
                    What is MetaMask?
                  </a>
                </p>,
                <div
                  className="landing-network-status"
                >
                  Select :
                  <p
                    className="rinkeby-network-label"
                  >
                    rinkeby test network
                  </p>
                </div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Please Login to ",
                    <a
                      href="https://metamask.io/"
                    >
                      MetaMask
                    </a>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Please Login to ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "MetaMask",
                      "href": "https://metamask.io/",
                    },
                    "ref": null,
                    "rendered": "MetaMask",
                    "type": "a",
                  },
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    href="https://www.cryptocompare.com/wallets/guides/how-to-use-metamask/"
                  >
                    What is MetaMask?
                  </a>,
                  "className": "landing_p",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "What is MetaMask?",
                    "href": "https://www.cryptocompare.com/wallets/guides/how-to-use-metamask/",
                  },
                  "ref": null,
                  "rendered": "What is MetaMask?",
                  "type": "a",
                },
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Select :",
                    <p
                      className="rinkeby-network-label"
                    >
                      rinkeby test network
                    </p>,
                  ],
                  "className": "landing-network-status",
                },
                "ref": null,
                "rendered": Array [
                  "Select :",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "rinkeby test network",
                      "className": "rinkeby-network-label",
                    },
                    "ref": null,
                    "rendered": "rinkeby test network",
                    "type": "p",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "className": "landing-logo",
            "src": "logo-large.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
